packages/core/src/messagingProtocols/mls/MLSService/MLSService.ts

```diff
@@ -924,8 +918,8 @@ export class MLSService extends TypedEventEmitter<Events> {
 
     this.dispatchNewCrlDistributionPoints(rotateBundle);
     // upload the clients public keys
-    if (!isCertificateRenewal) {
-      // we only upload public keys for the initial certification process. Renewals do not need to upload new public keys
+    if (!isMLSDevice(client)) {
+      // we only upload public keys for the initial certification process if the device is not already a registered MLS device.
```

Comment:
This is the important bit (not uploading key when the device already is an MLS device even if it was not enrolled)